File: dentsys-api/src/controllers/insurance-controller.js
import Insurance from "../models/insurance-model.js";

export default class InsuranceController {
    // POST
    static async addInsurance(req, res) {
        const data = req.body;
        try {
            const newInsuranceId = await Insurance.createInsurance(data);
            const newInsurance = await Insurance.getByInsuranceId(newInsuranceId)
            return res.status(201).json({ message: 'Insurance created successfully from controller', newInsurance});
        } catch (error) {
            res.status(500).json({ error: error.message });
        }
    }

    // GET
    static async getInsurance(req, res) {
        const data = req.params.id;
        try {
            const insurance = await Insurance.getInsurance(data);
            return res.status(200).json({ message: 'Insurance retrieved successfully from controller', insurance});
        } catch (error) {
            res.status(500).json({ error: error.message });
        }
    }

    // PUT
    static async updateInsurance(req, res) {
        const data = req.body;
        const patient_id = req.params.id;
        data.patient_id = patient_id;
        try {
            const updatedInsurance = await Insurance.updateInsurance(data);
            return res.status(200).json({ message: 'Insurance updated successfully from controller', updatedInsurance});
        } catch (error) {
            res.status(500).json({ error: error.message });
        }
    }


    // DELETE
    static async removeInsurance(req, res) {
        const data = req.params.id;
        try {
            const delInsurance = await Insurance.deleteInsurance(data);
            return res.status(200).json({ message: 'Insurance deleted successfully from controller', delInsurance});
        } catch (error) {
            res.status(500).json({ error: error.message });
        }
    }
}

File: dentsys_client/lib/controllers/insurance_controller.dart
import 'package:dentsys_client/models/insurance_model.dart';
import 'package:dentsys_client/services/insurance_service.dart';

class InsuranceController {
  final InsuranceService _insuranceService = InsuranceService();

  // POST
  Future<Insurance> createInsurance(Insurance insurance) async {
    try {
      final createdInsurance = await _insuranceService.addInsuranceService(insurance);
      print('created insurance from frontend controller: $createdInsurance');
      return createdInsurance;
    } catch (error) {
      throw Exception('Error adding insurance from frontend controller: $error');
    }
  }
  
  // GET

  // PUT
  Future<void> updateInsurance(Insurance insurance) async {
    try {
      await _insuranceService.updateInsuranceService(insurance);
    } catch (error) {
      throw Exception('Error updating insurance from frontend controller: $error');
    }
  }
}